tests/unit/test_signature.cpp

```diff
@@ -222,102 +178,341 @@ const std::vector<uint8_t> SignatureTest::_testMessageDigestSHA1{
+    {
+        std::make_shared<ECDSASignaturePrivateKeyCtx>(_validEccPrivateKey, DigestTypes::SHA256),
+        std::make_shared<ECDSASignaturePublicKeyCtx>(_validEccPublicKey, DigestTypes::SHA256),
+        {
+            0x30, 0x45, 0x02, 0x20, 0x79, 0x65, 0x52, 0xe1, 0xd0, 0x2e, 0x8e, 0x14, 0x16, 0xd4,
+            0x10, 0x29, 0x38, 0x30, 0x73, 0x5b, 0x49, 0x92, 0x00, 0x21, 0x9e, 0x60, 0x5c, 0x19,
+            0xb1, 0x03, 0x4e, 0x2d, 0x59, 0x8e, 0x64, 0xce, 0x02, 0x21, 0x00, 0xd5, 0x48, 0xb1,
+            0x91, 0x77, 0xe3, 0x20, 0x7a, 0xe5, 0x09, 0x0a, 0xdb, 0xa4, 0xab, 0xf0, 0x30, 0x9e,
+            0xaf, 0x34, 0xb6, 0xa2, 0xd1, 0x1f, 0xe1, 0x38, 0xf8, 0xf1, 0x41, 0x67, 0x3e, 0x81,
+            0x2a
+        }
+    },
+    // [10] ECDSA Signature using SHA512
+    {
+        std::make_shared<ECDSASignaturePrivateKeyCtx>(_validEccPrivateKey, DigestTypes::SHA512),
+        std::make_shared<ECDSASignaturePublicKeyCtx>(_validEccPublicKey, DigestTypes::SHA512),
+        {
+            0x30, 0x45, 0x02, 0x21, 0x00, 0xeb, 0x9b, 0x42, 0x71, 0xab, 0x1c, 0x5e, 0x02, 0x00,
+            0xe0, 0x17, 0x95, 0xf2, 0x47, 0x77, 0xf6, 0x5f, 0xc5, 0x91, 0xe5, 0xd2, 0x51, 0xde,
+            0xd9, 0x22, 0x9d, 0x4b, 0xab, 0x86, 0xe5, 0xc8, 0x70, 0x02, 0x20, 0x2f, 0xf2, 0x25,
+            0x48, 0x93, 0x12, 0xee, 0x9c, 0x65, 0x39, 0xb5, 0x14, 0x4a, 0x6d, 0x20, 0xea, 0x22,
+            0x46, 0x4a, 0x90, 0x7b, 0xca, 0xa9, 0x01, 0xf1, 0xb8, 0xae, 0xff, 0x05, 0x80, 0xd9,
+            0x3a
+        }
+    },
+    // [11] ECDSA Signature using SHA512 and using a certifcate for the cerification
+    {
+        std::make_shared<ECDSASignaturePrivateKeyCtx>(_validEccPrivateKey, DigestTypes::SHA512),
+        std::make_shared<ECDSASignaturePublicKeyCtx>(_validEccCertificate, DigestTypes::SHA512),
+        {
+            0x30, 0x45, 0x02, 0x21, 0x00, 0x94, 0x9a, 0x85, 0xda, 0xc4, 0x45, 0xcf, 0x28, 0xfb,
+            0x79, 0x38, 0x9b, 0xf0, 0x1c, 0xc8, 0x77, 0x54, 0x7e, 0x86, 0x5c, 0x8c, 0x65, 0x5a,
+            0x3f, 0x7b, 0x0f, 0x83, 0x5a, 0xd2, 0x65, 0x0d, 0x38, 0x02, 0x20, 0x2f, 0x9f, 0x3a,
+            0x2d, 0x14, 0xd9, 0x45, 0xd0, 0x88, 0x48, 0x40, 0x3d, 0x2b, 0x7e, 0x51, 0x6e, 0x37,
+            0x61, 0xe8, 0xc8, 0xfe, 0xcb, 0xb7, 0x49, 0xc0, 0xc2, 0x64, 0x92, 0x46, 0x51, 0x77,
+            0xc7
+        }
+    }
 
-    auto ctx256 = RSASignaturePrivateKeyCtx(key, _sha256PKCSSignaturePadding);
-    EXPECT_NO_THROW(signature = ctx256.signDigest(_testMessageDigestSHA256));
-    EXPECT_EQ(_validPKCS1SignatureSHA256, signature);
+};
 
-    auto ctx512 = RSASignaturePrivateKeyCtx(key, _sha512PKCSSignaturePadding);
-    EXPECT_NO_THROW(signature = ctx512.signDigest(_testMessageDigestSHA512));
-    EXPECT_EQ(_validPKCS1SignatureSHA512, signature);
-}
+const std::vector<uint8_t>& SignatureTest::_validPKCS1SignatureSHA256 = SignatureTestData[1].validSignature;
+const std::vector<uint8_t>& SignatureTest::_validPKCS1SignatureSHA512 = SignatureTestData[2].validSignature;
```

Comment:
I couldn't find a use of `_validPKCS1SignatureSHA512`. Am I missing something or should it be removed?